devito/passes/clusters/blocking.py

```diff
@@ -1,12 +1,45 @@
 from collections import Counter
 
 from devito.ir.clusters import Queue
-from devito.ir.support import TILABLE, IntervalGroup, IterationSpace
+from devito.ir.support import (SEQUENTIAL, SKEWABLE, TILABLE, Interval, IntervalGroup,
+                               IterationSpace)
 from devito.symbolics import uxreplace
-from devito.tools import timed_pass
 from devito.types import IncrDimension
 
-__all__ = ['Blocking']
+from devito.symbolics import xreplace_indices
+
+__all__ = ['blocking']
+
+
+def blocking(clusters, options):
+    """
+    Loop blocking to improve data locality.
+
+    Parameters
+    ----------
+    clusters : tuple of Clusters
+        Input Clusters, subject of the optimization pass.
+    options : dict
+        The optimization options.
+        * `blockinner` (boolean, False): enable/disable loop blocking along the
+           innermost loop.
+        * `blocklevels` (int, 1): 1 => classic loop blocking; 2 for two-level
+           hierarchical blocking.
+        * `skewing` (boolean, False): enable/disable loop skewing.
+
+    Notes
+    ------
+    In case of skewing, if 'blockinner' is enabled, the innermost loop is also skewed.
+    """
+    processed = preprocess(clusters, options)
```

Comment:
this was pointed out a few times, "processed" is only at the end, when it needs no more processing... anyway, nitpicking